template/java8/model/src/main/java/com/openfaas/model/Request.java

```diff
@@ -4,16 +4,28 @@
 package com.openfaas.model;
 
 import java.util.Map;
+import java.util.HashMap;
+import java.net.URLDecoder;
+import java.io.UnsupportedEncodingException;
 
 public class Request implements IRequest {
 
     private Map<String, String> headers;
     private String body;
+    private Map<String, String> queryParameters;
+    private String queryRaw;
 
     public Request(String body, Map<String, String> headers) {
         this.body = body;
         this.headers = headers;
     }
+    
+    public Request(String body, Map<String, String> headers,String queryRaw) {
+        this.body = body;
+        this.headers = headers;
+        this.queryRaw = queryRaw;
+        this.queryParameters = this.parseQueryParameters();
+    }
 
     public String getBody() {
```

Comment:
Does it need `@Override`?

Reply:
@Override is not needed! Its good to have on methods to indicate that method is overridden from super type.

Reply:
Maybe we could add it?